src/main/java/org/gridsuite/explore/server/services/StudyService.java

```diff
@@ -6,138 +6,117 @@
-import reactor.core.publisher.Mono;
-import reactor.core.scheduler.Schedulers;
 
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
-import java.util.logging.Level;
+import java.io.IOException;
+import java.util.*;
 import java.util.stream.Collectors;
 
+import static org.gridsuite.explore.server.ExploreException.Type.*;
+
 /**
  * @author Etienne Homer <etienne.homer at rte-france.com>
  */
 @Service
 public class StudyService implements IDirectoryElementsService {
-    private static final String ROOT_CATEGORY_REACTOR = "reactor.";
-
     private static final String STUDY_SERVER_API_VERSION = "v1";
 
     private static final String DELIMITER = "/";
-
-    private final WebClient webClient;
+    private final RestTemplate restTemplate;
     private String studyServerBaseUri;
 
     @Autowired
-    public StudyService(@Value("${backing-services.study-server.base-uri:http://study-server/}") String studyServerBaseUri,
-                            WebClient.Builder webClientBuilder) {
+    public StudyService(@Value("${backing-services.study-server.base-uri:http://study-server/}") String studyServerBaseUri, RestTemplate restTemplate) {
         this.studyServerBaseUri = studyServerBaseUri;
-        this.webClient = webClientBuilder.build();
+        this.restTemplate = restTemplate;
     }
 
     public void setStudyServerBaseUri(String studyServerBaseUri) {
         this.studyServerBaseUri = studyServerBaseUri;
     }
 
-    public Mono<Void> insertStudyWithExistingCaseFile(UUID studyUuid, String userId, UUID caseUuid, Map<String, Object> importParams) {
+    public void insertStudyWithExistingCaseFile(UUID studyUuid, String userId, UUID caseUuid, Map<String, Object> importParams) {
         String path = UriComponentsBuilder.fromPath(DELIMITER + STUDY_SERVER_API_VERSION +
                         "/studies/cases/{caseUuid}?studyUuid={studyUuid}")
                 .buildAndExpand(caseUuid, studyUuid)
                 .toUriString();
-
-        return webClient.post()
-                .uri(studyServerBaseUri + path)
-                .header(HEADER_USER_ID, userId)
-                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON.toString())
-                .body(importParams != null ? BodyInserters.fromValue(importParams) : BodyInserters.empty())
-                .retrieve()
-                .bodyToMono(Void.class)
-                .publishOn(Schedulers.boundedElastic())
-                .log(ROOT_CATEGORY_REACTOR, Level.FINE);
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        headers.add(HEADER_USER_ID, userId);
+        HttpEntity<Map<String, Object>> request = new HttpEntity<>(
+                importParams, headers);
+        try {
+            restTemplate.exchange(studyServerBaseUri + path, HttpMethod.POST, request, Void.class);
+        } catch (HttpStatusCodeException e) {
+            throw new ExploreException(INSERT_STUDY_FAILED);
```

Comment:
remove this try catch that was not here before to handle the exception with the exception handler ?

Reply:
done